crates/signer/src/bls/handlers.rs

```diff
@@ -359,6 +380,181 @@ where
         }
     }
 
+    fn handle_partial_from_network(
+        &self,
+        partial: PartialSignatureWithRequest<BLS>,
+        sender: u16,
+        new_message_to_sign: &UnboundedSender<BlsSignatureRequest>,
+    ) {
+        let PartialSignatureWithRequest { sig, req } = partial;
+
+        Metrics::report_partials_received(1);
+
+        let Some(stored_req) = self.try_parse_req(req.clone()) else {
+            tracing::warn!(sender_id = sender, app = ?req.args.app(), alg = ?req.alg, "Received partial with unsupported app");
+            return;
+        };
+
+        // Verify the validity of the partial signature for the specified id
+        match self.try_verify(&req.m, &stored_req.dst, sig, &sender, &req.alg) {
+            Ok(true) => (),
+            Ok(false) => {
+                tracing::error!(sender_id = sender, "Received invalid partial signature");
+                Metrics::report_invalid_partials(1);
```

Comment:
and it _might_ be worth reporting the request_id here, under the assumption that it should very rarely happen it'd be good to know/to repair it

Reply:
though actually maybe just knowing the time is sufficient because presumably the reuqest will still be outstanding onchain